src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -588,15 +585,23 @@ export class NotebookService extends Disposable implements INotebookService {
 			// 4. Notebook is trusted. Don't need to save state of untrusted notebooks
 			let notebookUriString = notebookUri.toString();
 			if (changeType === NotebookChangeType.Saved && firstIndex(this._trustedCacheQueue, uri => uri.toString() === notebookUriString) < 0) {
-				// Only save if it's trusted
-				let notebook = find(this.listNotebookEditors(), n => n.id === notebookUriString);
-				if (notebook && notebook.model) {
-					if (notebook.model.trustedMode) {
-						this._trustedCacheQueue.push(notebookUri);
-					} else {
-						this._unTrustedCacheQueue.push(notebookUri);
+				if (isTrusted) {
+					this._trustedCacheQueue.push(notebookUri);
+					await this.updateTrustedCache();
+				} else if (isTrusted === false) {
+					this._unTrustedCacheQueue.push(notebookUri);
+					await this.updateTrustedCache();
+				} else {
+					// Only save as trusted if the associated notebook model is trusted
+					let notebook = find(this.listNotebookEditors(), n => n.id === notebookUriString);
+					if (notebook && notebook.model) {
+						if (notebook.model.trustedMode) {
```

Comment:
We would overwrite that entry in the trusted cache, but it would still remain trusted. This code is already present, but has moved around with this change.